### DISCUSSION TITLE
Mod settings aren't immediately visible on mod icons

### DISCUSSION THREAD

**Discussion by Akosih97** [ppy/osu#30883] [Ideas] [answered]
### Type

Game behaviour

### Bug description

I played todays beatmap of the day, which had a mandatory HD modifier and used a DT x1.2 mod on top of that. The hidden modifier worked partially: The enclosing circles around the hit circles were missing, however the hit circles (and sliders) did not disappear before I had to hit them, which made the HD mod significantly easier (I did not submit a score). The bug reoccured when restarting the beatmap after dying. Previously on the same day, with the same version and a lower DT x1.15 mod the bug did not occur. 

Unfortunately the bug occured just 5 minutes before the beatmap of the day was closing so I had no time to test it thoroughly. Therefore I could not check if it was still there after restarting the game. Also taking a screenshot was not possible well, because I had to stop playing and then died before the shot was taken. Maybe you can see it below anyways: While being on the slider, the following hit circles which should have already disappeared are still visible.

After restarting the game, using HD modifiers on other beatmaps and the new beatmap of the day worked fine.

### Screenshots or videos

![Screenshot from 2024-11-26 00-56-54](https://github.com/user-attachments/assets/7bd2fbdd-068e-4181-b257-54d2239b028c)

### Version

2024.1115.3-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17910349/compressed-logs.zip)

**Comment by Akosih97** [comment]
I took a short video of the beginning of the song using my phone, which clearly shows the bug. In case this is helpful I can provide it.

**Comment by Ladwick** [comment]
I think you are talking about hidden with the 'only fade approach circles' option enabled maybe? Try the map again in solo, enable hidden and toggle said option on in the customize dropdown menu, sorry if I'm misunderstanding your problem

**Comment by Akosih97** [comment]
> I think you are talking about hidden with the 'only fade approach circles' option enabled maybe? Try the map again in solo, enable hidden and toggle said option on in the customize dropdown menu, sorry if I'm misunderstanding your problem

I think you are correct. Sorry, I didnt know this customization option existed. I did not change it deliberately at least. In this case, HD was mandatory for the daily beatmap. Do you know if it is possible to customize a mandatory mod of a daily beatmap? If not, then it changed on its own due to some bug.

**Comment by Ladwick** [comment]
Glad I was able to help, and yes, it happened already once in daily challenges, I don't remember which one though, but you can check it hovering the HD symbol in the beatmap card of the daily screen and the tooltip will say the option is on.

**Comment by frenzibyte** [comment]
I think we can improve the appearance of the icon to clear confusion here, I'm certain this is not the first time someone thinks this is a bug.

**Comment by Akosih97** [comment]
> Glad I was able to help, and yes, it happened already once in daily challenges, I don't remember which one though, but you can check it hovering the HD symbol in the beatmap card of the daily screen and the tooltip will say the option is on.

So you are saying the option was active for the whole day last time? Because the HD mod was definitely the "usual" one a few hours ago when I already played the map and spontaneously changed.

**Comment by frenzibyte** [comment]
Today's daily challenge just started, what you saw a few hours ago was a previous challenge.

**Comment by Akosih97** [comment]
> Today's daily challenge just started, what you saw a few hours ago was a previous challenge.

I am just talking about the challenge from 2024-11-25 (COMA Atm's Insane). When I set my score 6hours ago the HD mod for me was definitely the usual one without the "only fade approach circles" option. And about one hour ago, when I tried to improve upon the score the "only fade approach circles" option was active.

I am very sure of this because I have never seen this option on the HD mod before the play 1h ago. Despite that, the score I set 6hours ago https://osu.ppy.sh/scores/3918930610 is actually saved in the database with the mod "HD (Only fade approach circles: On)". 

So either there was a bug which led to the "only fade approach circles" option not being activated during my tries on the map 6h ago. Or it was somehow possible to change this option for the daily map, which probably should not be the case. (And even if it should have been possible there would be a mismatch between the mod which was actually active during my game and the mod information of the online score)

**Comment by frenzibyte** [comment]
The only possible case here is that the option did not work in your previous attempts for some reason, but there's really nothing to work on based on that alone. If you find a method to replicate this bug reliably, then send an email to pe@ppy.sh with the details you have found and action will be taken from there.

**Comment by peppy** [comment]
This applies to basically every mod / mod setting. I've made this a discussion as it needs design and even UX considerations.

In the past we discussed adding a small indicator light or something to the icons. That hasn't happened yet.

cc @ppy/team-design

**Reply by cl8n** [reply]
can do, that was something I was gonna look into regardless before I got roped into... every other design thing :sweat_smile: 

I'm a little confused about this discussion though, it seems like this is reporting a bug with daily challenge and it wouldn't be resolved by a design change

**Reply by Akosih97** [reply]
> can do, that was something I was gonna look into regardless before I got roped into... every other design thing :sweat_smile: 
> 
> I'm a little confused about this discussion though, it seems like this is reporting a bug with daily challenge and it wouldn't be resolved by a design change

Sry, that is due to a lot of confusion on my part. I did not know the HD mod option "only fade approach circles" existed and indeed also experienced a bug in the daily challenge, where for my first few plays of the daily the "only fade approach circles" option did not apply to me even though it should have. And then on the later playthrough it worked which I misinterpreted as a bug because I did not know the option existed.

Thats why it was proposed to change the icons appearance in order to make inexperienced lazer users (like me) notice that there is a difference.

This is a separate problem from the bug I experienced though. And unfortunately I have no idea how to reproduce the bug as it happened for this very specific daily challenge with the mandatory HD (only fade approach circles) mod.

**Reply by peppy** [reply]
>  where for my first few plays of the daily the "only fade approach circles" option did not apply to me even though it should have

you probably just noticed the first object always having an approach circle. that's a user setting.

> I'm a little confused about this discussion though, it seems like this is reporting a bug with daily challenge and it wouldn't be resolved by a design change

i still think this is a valid concern that should be addressed at a mod icon level, and applied visually everywhere.

**Reply by Joehuu** [reply, marked as answer]
Will be addressed in https://github.com/ppy/osu/pull/32970.
